### DISCUSSION TITLE
zxing-android-embedded and CameraX

### DISCUSSION THREAD

**Discussion by kynphlee** [journeyapps/zxing-android-embedded#712] [General]
Has anyone used zxing-android-embedded with Android's CameraX framework? I was looking at the ContinuousCaptureActivity sample and was wondering if there would have to be changes to make it work with CameraX or if zxing-android-embedded supports it.

**Comment by rkistner** [comment]
The CameraX framework has a couple of advantages, at least in theory. For a start, the underlying Camera2 API supports things like separating image streams for preview and scanning. And the CameraX wrappers around that may remove some of the complexity around camera management that we currently have.

There is no support right now - it would require significant changes to the existing code. And it would likely be a complete switch, not a new option. I'm not aware of anyone that has tried it yet.
